dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/data/upload/v2/DataUploadRunnableTest.kt

```diff
@@ -90,17 +90,24 @@ internal class DataUploadRunnableTest {
     @Forgery
     lateinit var fakeDataUploadConfiguration: DataUploadConfiguration
 
+    private var expectedBatchesHandled: Int = 0
+
     private lateinit var testedRunnable: DataUploadRunnable
 
     @BeforeEach
     fun `set up`(forge: Forge) {
+        // to make sure the basic tests are not flaky
+        // the cases where the maxBatchesPerUploadJob is variable are tested in dedicated test
+        // methods
+        fakeDataUploadConfiguration = fakeDataUploadConfiguration.copy(maxBatchesPerUploadJob = 1)
```

Comment:
where the flakiness comes from? In fact it seems we are always testing the case where `maxBatchesPerUploadJob = = 1`, because I don't see any place where it would have a different value

Reply:
maybe it's a mistake on my end I will update those tests where it should have a different value. I wanted to make sure that the basics tests that are testing the increase/decrease interval are only based on a single value to avoid problems and complexity. I will add a better explanation